### PR TITLE
Decorate Class PR

### PR DESCRIPTION
**Pull request by kkasaija**
- Added base Decorator
- Added capitalize decorator
- Added trimmer decorator

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+
+# Person class
+class Person < Nameable
   attr_accessor :name, :age
   attr_reader :id
 
-  def initialize(age, parent_permission: true, name: 'Unknown')
+  def correct_name
+    @name
+  end
```
Comment (Mihndim2020) [inline person.rb:15]: - [OPTIONAL] The `correct_name` method should come after the initialize method, please kindly fix that.

File: README.md [hunk 2]
```diff
@@ -99,7 +99,7 @@ After you're finished please remove all the comments and instructions!
 
 > Describe between 1-3 key features of the application.
 
-- **[Person class]**
+- **[Person class with Decorators]**
 - **[Student class]**
 - **[Teacher class]**
```
Comment (Mihndim2020) [inline README.md:104]: - [OPTIONAL] It will be nice to update your getting started section with instructions on how to clone and run your project locally.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
### Hi @Kasaija-Kenneth, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (kkasaija) [reply person.rb:15]: Ok, Noted with thanks. I am correcting it immediately @Mihndim2020

File: README.md (hunk 2 above)
Comment (kkasaija) [reply README.md:104]: This is noted

**Pull request merged**